app/controllers/fhir/r4/api_controller.rb

```diff
@@ -342,12 +335,100 @@ def create
                             created_by: @current_actor_label,
                             comment: "New vaccine added via API (ID: #{resource.id}).")
       end
+    when 'observation'
+      return if doorkeeper_authorize!(*OBSERVATION_WRITE_SCOPES)
+
+      resource, fhir_map = build_laboratory(contents)
+
+      ActiveRecord::Base.transaction do
+        unless referenced_patient_valid_for_client?(resource, :patient_id) && resource.save(context: :api) && fhir_map.all? { |_k, v| v[:errors].blank? }
+          req_json = JSON.parse(request.body.string)
+          status_unprocessable_entity(resource, fhir_map, req_json) && (raise SaveError)
```

Comment:
I went to error raising because of this deprecation warning when I use return:
![image](https://user-images.githubusercontent.com/52747882/118139118-0b10fa00-b3d5-11eb-984a-43cc33a6607c.png)

I was pretty confused why this warning only pops up for the new code I add, but from what I can tell, rails is smart enough to know the difference between old code and new code, but then I also don't understand why it wouldn't be helpful to give the deprecation warning even for the old code that uses `return`. Either way I was planning to move everything to `raise` when I refactor the API controller. I considered doing it here, but it felt like it might complicate the diff.

Reply:
Oh interesting I have not seen that. Sounds like a good rule of thumb anyway. I agree no need to do it here, especially since you can easily take care of that in the refactor task you're doing, SARAALERT-1464.